### PR TITLE
[Enhancement] Support higher version of Hive metastore

### PR DESCRIPTION
**Pull request by Smith-Cruise**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Problem Summary(Required) ：

The previous pr #19214 changed too much, and introduce a new parameter, it's not compatible with users which use Hive 1.x.

This pr is simpler, it will try to use `get_table()` first, if failed, use `get_table_req()` instead.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 11 (45.45%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetaStoreThriftClient.java|5|11|45.45%|[624, 625, 626, 627, 628, 629]|

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetaStoreThriftClient.java [hunk 1]
```diff
@@ -596,7 +618,16 @@ public Table getTable(String dbName, String tableName) throws MetaException, TEx
 
     @Override
     public Table getTable(String catName, String dbName, String tableName) throws MetaException, TException {
-        return client.get_table(dbName, tableName);
+        try {
+            // Using get_table_req() first, if user is using Hive v1.x version, it will request failed,
+            // then fail over to use get_table() instead.
+            GetTableRequest req = new GetTableRequest(dbName, tableName);
+            req.setCapabilities(version);
+            return client.get_table_req(req).getTable();
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetaStoreThriftClient.java:626]: why not use `get_table` at first

Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetaStoreThriftClient.java:626]: Most people are using Hive version `>= 2.x`, which already supports `get_table_req()`.

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetaStoreThriftClient.java:626]: use get_table first +1

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Smith-Cruise** [comment]
run starrocks_be_unittest

**Comment by Smith-Cruise** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 11 (45.45%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetaStoreThriftClient.java|5|11|45.45%|[624, 625, 626, 627, 628, 629]|

**Comment by Smith-Cruise** [comment]
@Mergify backport branch-3.0

**Pull request merged**